src/platform_impl/windows/system_tray.rs

```diff
@@ -118,30 +114,27 @@ impl SystemTrayBuilder {
         )));
       }
 
-      let app_system_tray = SystemTray { hwnd, hmenu };
-      app_system_tray.set_icon_from_buffer(&self.icon, 32, 32);
+      let system_tray = SystemTray { hwnd };
+      system_tray.set_icon_from_buffer(&self.icon, 32, 32);
 
-      // system tray handler
+      // system_tray event handler
       let event_loop_runner = window_target.p.runner_shared.clone();
-      let menu_handler = MenuHandler::new(
-        Box::new(move |event| {
+      let traydata = TrayLoopData {
+        hmenu,
+        sender: Box::new(move |event| {
           if let Ok(e) = event.map_nonuser_event() {
             event_loop_runner.send_event(e)
           }
         }),
-        MenuType::ContextMenu,
-        None,
+      };
+      commctrl::SetWindowSubclass(
+        hwnd,
+        Some(tray_subclass_proc),
+        TRAY_SUBCLASS_ID,
+        Box::into_raw(Box::new(traydata)) as _,
```

Comment:
I just realize there's no `from_raw` for `traydata` and `menu_handler`.
This may cause some memory leaks. Maybe add these to the tray type and release them in drop trait?

Reply:
Would it be enough to just use `Box::from_raw()` ?

Reply:
`Box::into_raw` will return a pointer and the owner of the type is sort of ignored by the borrow checker.
This pointer won't be released on `WM_DESTROY` I suppose. I tried `free` from libc on macos before and that didn't work
The safest way is `from_raw` and make the type drop out of scope.

Reply:
We are already doing something similar with `Box::from_raw`, see https://github.com/tauri-apps/tao/blob/999a2cc524a2ba2f2126f72ec4f88774d61b008c/src/platform_impl/windows/event_loop.rs#L844

Reply:
yes something like this will work.

Reply:
alright, I just pushed a commit and let all `WM_DESTORY` execute where I call `Box::from_raw`